src/main/java/uk/gov/ch/developer/docs/controller/developer/UserCallbackController.java

```diff
@@ -1,36 +1,38 @@
 package uk.gov.ch.developer.docs.controller.developer;
 
-import java.net.URI;
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.MediaType;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.reactive.function.BodyInserters;
-import org.springframework.web.reactive.function.client.WebClient;
 import com.nimbusds.jose.Payload;
 import net.minidev.json.JSONObject;
-import reactor.core.publisher.Mono;
+import uk.gov.ch.developer.docs.DocsWebApplication;
 import uk.gov.ch.oauth.IIdentityProvider;
 import uk.gov.ch.oauth.IOauth;
-import uk.gov.ch.oauth.OAuthToken;
+import uk.gov.ch.oauth.UserProfileResponse;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.handler.SessionHandler;
 
 @Controller
 @RequestMapping("${callback.url}")
 public class UserCallbackController {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DocsWebApplication.APPLICATION_NAME_SPACE);
+
     @Autowired
-    IIdentityProvider identityProvider;
+    private IIdentityProvider identityProvider;
+
     @Autowired
     private IOauth oauth;
 
     @GetMapping
     public String getCallback(@RequestParam("state") String state,
-            @RequestParam("code") String code) {
+            @RequestParam("code") String code, final HttpServletRequest httpServletRequest) {
```

Comment:
Should the serverlet request have an annontation?

Reply:
What annotation?

Reply:
Good question... as it is already auto wired, I'm not sure.